packages/shared/src/contexts/AuthContext.tsx

```diff
@@ -120,7 +122,29 @@ export const AuthContextProvider = ({
 }: AuthContextProviderProps): ReactElement => {
   const [loginState, setLoginState] = useState<LoginState | null>(null);
   const endUser = user && 'providers' in user ? user : null;
-  const referral = user?.referrer;
+  const referralCampaign = useMemo<{
+    referralId: string;
+    referralOrigin: string;
+  }>(() => {
+    const cookie = typeof window !== 'undefined' ? getCookieObject() : {};
+    const { join_referral: joinReferralCookie } = cookie as Partial<{
+      join_referral: string;
+    }>;
+
+    if (!joinReferralCookie) {
+      return undefined;
+    }
+
+    const [referralId, referralOrigin] = joinReferralCookie.split(':');
+
+    return {
+      referralId,
+      referralOrigin,
+    };
```

Comment:
Let's turn it into a standalone function. Will look much better

Reply:
Another thought about this, shouldn't we fetch only when needed? Instead of eagerly loading we can load it only when needed. Hence, ensuring window is defined and everything is there already

Reply:
Refactored to use boot.